src/context.rs

```diff
@@ -44,4 +52,27 @@ impl Context {
 
         Ok(::device::new(self, ptr))
     }
+
+    /// Creates a device for a given major/minor device number pair
+    pub fn device_from_devnum(&self, kind: ::DeviceType, major: u64, minor: u64) -> ::Result<Device> {
+        let devnum = makedev(major, minor);
+        let ptr = try_alloc!(unsafe {
+            ::ffi::udev_device_new_from_devnum(self.udev, match kind {
+                ::DeviceType::Char => b'c' as ::libc::c_char,
+                ::DeviceType::Block => b'b' as ::libc::c_char
+            }, devnum)
+        });
+
+        Ok(::device::new(self, ptr))
+    }
+
+    /// Creates a device for from a subsystem and sysname
+    pub fn device_from_subsystem_sysname<T: AsRef<OsStr>, U: AsRef<OsStr>>(&self, subsystem: T, sysname: U) -> ::Result<Device> {
+        let subsystem = try!(::util::os_str_to_cstring(subsystem));
+        let sysname = try!(::util::os_str_to_cstring(sysname));
```

Comment:
If libudev library is required to be at least systemd v228 then there is nothing extra to be done here. Otherwise, it is probably advisable to correct for a bug, now fixed: https://github.com/systemd/systemd/commit/a9ec9f29420623133c419ddc8864921a824751fb. Personally, I feel that this library and Rust are new enough that requiring systemd v228 seems like a fine idea to me.